packages/website/docs/using-react.md

```diff
@@ -0,0 +1,127 @@
+---
+id: using-react
+title: Using Autocomplete with React
+---
+
+Learn how to create and use an React Autocomplete component.
+
+This guide shows how to create a React Autocomplete component. It uses the [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) and [`useEffect`](https://reactjs.org/docs/hooks-reference.html#useeffect) hooks to create and mount the component. It doesn't define specific [sources](sources). Rather, you can pass [sources](sources) and other [options](basic-options) as [props](https://reactjs.org/docs/components-and-props.html).
+
+## Prerequisites
+
+This tutorial assumes that you have:
+- an existing [React (v 16.8+) application](https://reactjs.org/docs/getting-started.html) where you want to implement the autocomplete dropdown
+- familiarity with the [basic Autocomplete configuration options](basic-options)
+
+## Creating the component
+
+Start with some boilerplate for creating a React component. This component uses the [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) hook to create a mutable ref object, `containerRef`, to mount the autocomplete on. To learn more about this hook, check out the [`useRef` React documentation](https://reactjs.org/docs/hooks-reference.html#useref).
+
+ All that you need to render is a `div` with the `containerRef` as the `ref`.
+
+```jsx title="Autocomplete.jsx"
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+### Mounting the autocomplete
+
+Now that you have access to the DOM through the `containerRef` object, you can create and mount the autocomplete instance.
+
+You can use the same [Autocomplete options](basic-options), but you must also pass the `renderer` and `render` parameters. You can rely on `props` to pass any autocomplete options you want to remain configurable.
```

Comment:
Wdyt of just adding a sentence that explains why?

```suggestion
You can use the same [Autocomplete options](basic-options), but you must also pass the `renderer` and `render` parameters. Those allow to plug the internal React mechanism into Autocomplete. You can rely on `props` to pass any autocomplete options you want to remain configurable.
```

Reply:
Good idea! Could you help explain a bit more about what the "React mechanism" is? I think this language is a bit vague, so it might not help the reader understand more.

Reply:
The default Autocomplete implementation is to use Preact's version of [`createElement`](https://preactjs.com/guide/v10/api-reference/#h--createelement), [`Fragment`](https://preactjs.com/guide/v10/api-reference/#fragment) and [`render`](https://preactjs.com/guide/v10/api-reference/#render).

Each framework has its own specifities for these APIs, and they work together as a whole (we can't mix them). Although the concepts are the same, the expected input and format differ slightly.

This is why we need to provide the React version of [`createElement`](https://reactjs.org/docs/react-api.html#createelement), [`Fragment`](https://reactjs.org/docs/react-api.html#reactfragment) and [`render`](https://reactjs.org/docs/rendering-elements.html#rendering-an-element-into-the-dom).

Once these values are specified, Autocomplete will use them to render the views.

Reply:
How does this sound:
Maybe it doesn't need to be more exact. How does this sound:

```suggestion
You can use the same [Autocomplete options](basic-options), but you must also pass the `renderer` and `render` parameters. These parameters let the component plug into internal React mechanisms. You can rely on `props` to pass any autocomplete options you want to remain configurable.
```

Reply:
Oops, missed the previous reply ☝️

Reply:
https://github.com/algolia/autocomplete.js/pull/442/commits/1332818d65852fb4fa9ea5d273b592cca842ab6c